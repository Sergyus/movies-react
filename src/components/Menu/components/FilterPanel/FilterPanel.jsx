import React from "react";
import PropTypes from "prop-types";
import styles from "./FilterPanel.scss";
import FilterPanelButton from "./components/FilterPanelButton";
import SelectContainer from "./components/Select";

const FilterPanel = ({
  fetchTopRatedMovies,
  fetchLatestMovies,
  fetchUpcomingMovies,
  changeFilter,
  filter
}) => {
  const setActiveItem = itemName =>
    itemName === filter ? styles.btn__active : null;
  return (
    <nav>
      <ul>
        <FilterPanelButton
          activeItem={filter}
          setActiveItem={setActiveItem}
          name="Актуальные"
          setActive={changeFilter}
          fetchMovies={fetchLatestMovies}
        />
        <FilterPanelButton
          activeItem={filter}
          setActiveItem={setActiveItem}
          name="Самые популярные"
          setActive={changeFilter}
          fetchMovies={fetchTopRatedMovies}
        />
        <FilterPanelButton
          activeItem={filter}
          setActiveItem={setActiveItem}
          name="Скоро будет"
          setActive={changeFilter}
          fetchMovies={fetchUpcomingMovies}
        />
        <SelectContainer
          setActiveItem={setActiveItem}
          setActive={changeFilter}
        />
      </ul>
    </nav>
  );
};

FilterPanel.propTypes = {
  fetchTopRatedMovies: PropTypes.func,
  fetchLatestMovies: PropTypes.func,
  fetchUpcomingMovies: PropTypes.func,
  changeFilter: PropTypes.func,
  filter: PropTypes.string
};

FilterPanel.defaultProps = {
  fetchTopRatedMovies: () => {},
  fetchLatestMovies: () => {},
  fetchUpcomingMovies: () => {},
  changeFilter: () => {},
  filter: "Актуальные"
};

export default FilterPanel;
