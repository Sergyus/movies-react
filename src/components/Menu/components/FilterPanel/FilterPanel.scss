@import "../../../../styles/variables";

.navItem {
  display: inline-block;
  margin-right: 15px;
  position: relative;
}

.btn {
  box-sizing: border-box;
  font-size: 1rem;
  border: none;
  outline: none;
  background-color: transparent;
  color: $blue-color;
  cursor: pointer;
  padding-bottom: 10px;
  font-family: "Helvetica Neue", Arial, Helvetica, sans-serif;
  font-weight: bold;

  &:hover {
    color: darken($blue-color, 10%);
  }

  &__active {
    border-bottom: 3px solid $blue-color;
    padding-bottom: 7px;
  }
}

.selectGenre {
  box-sizing: border-box;
  width: 135px;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  outline: none;
  border: none;
  cursor: pointer;
  color: $blue-color;
  padding-bottom: 10px;
  font-size: 1rem;
  font-family: "Helvetica Neue", Arial, Helvetica, sans-serif;
  font-weight: bold;
  background: url("../../../../assets/images/sprites.png") -130px -102px no-repeat;

  &:hover {
    color: darken($blue-color, 10%);
  }

  &.btn__active {
    border-bottom: 3px solid $blue-color;
    padding-bottom: 7px;
  }
}
