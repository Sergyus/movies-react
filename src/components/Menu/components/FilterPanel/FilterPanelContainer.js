import { connect } from 'react-redux';
import FilterPanel from './FilterPanel';
import {
  fetchTopRatedMovies, fetchLatestMovies, fetchUpcomingMovies, changeFilter,
  fetchedGenresSelector, filterSelector,
} from '../../../../modules/app';

const mapStateToProps = state => ({
  fetchedGenres: fetchedGenresSelector(state),
  filter: filterSelector(state),
});

const actions = {
  fetchTopRatedMovies,
  fetchLatestMovies,
  fetchUpcomingMovies,
  changeFilter,
};

export default connect(mapStateToProps, actions)(FilterPanel);
