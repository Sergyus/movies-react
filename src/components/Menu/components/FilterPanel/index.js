import FilterPanelContainer from './FilterPanelContainer';

export default FilterPanelContainer;
