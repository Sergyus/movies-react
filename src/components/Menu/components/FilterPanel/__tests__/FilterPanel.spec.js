import React, { useState } from 'react';
import { create, act } from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import thunkMiddleware from 'redux-thunk';
import { Provider } from 'react-redux';
import FilterPanel from '../FilterPanel';

global.fetch = require('jest-fetch-mock');

describe('FilterPanel component', () => {
  const middlewares = [thunkMiddleware];
  const mockStore = configureStore(middlewares);
  const initialState = {
    app: {
      movies: [],
      genres: [],
      isLoading: true,
      currentPage: 1,
      error: false,
      filter: 'Trending',
      currentGenre: '',
      currentSearchQuery: '',
      videoKey: '',
      currentTotalPages: 1,
    },
    ui: {
      layout: 'grid',
    },
  };
  const store = mockStore(initialState);
  const mockSetActive = jest.fn();
  it('FilterPanel component renders correctly', () => {
    const mockFetchTopRatedMovies = jest.fn();
    const mockFetchLatestMovies = jest.fn();
    const mockFetchUpcomingMovies = jest.fn();
    const mockchangeFilter = jest.fn();
    const MockComponent = () => {
      const [activeItem, setActiveItem] = useState('Top Rated');
      return (
        <FilterPanel
          activeItem={activeItem}
          setActiveItem={setActiveItem}
          setActive={mockSetActive}
          fetchTopRatedMovies={mockFetchTopRatedMovies}
          fetchLatestMovies={mockFetchLatestMovies}
          fetchUpcomingMovies={mockFetchUpcomingMovies}
          changeFilter={mockchangeFilter}
        />
      );
    };
    const tree = create(
      <Provider store={store}>
        <MockComponent />
      </Provider>,
    );
    expect(tree.toJSON()).toMatchSnapshot();
  });
  it('FilterPanel component without props works correctly', () => {
    const MockComponent = () => {
      const [activeItem, setActiveItem] = useState('Top Rated');
      return (
        <FilterPanel
          activeItem={activeItem}
          setActiveItem={setActiveItem}
          setActive={mockSetActive}
        />
      );
    };
    const tree = create(
      <Provider store={store}>
        <MockComponent />
      </Provider>,
    );
    const latestBtn = tree.root.findByProps({ 'aria-label': 'Trending' });
    const topRatedBtn = tree.root.findByProps({ 'aria-label': 'Top Rated' });
    const upcomingBtn = tree.root.findByProps({ 'aria-label': 'Coming Soon' });
    const genresSelect = tree.root.findByProps({ 'aria-label': 'Genres' });
    const mockFetchTopRatedMovies = jest.fn(latestBtn.props.onClick);
    const mockFetchLatestMovies = jest.fn(topRatedBtn.props.onClick);
    const mockFetchUpcomingMovies = jest.fn(upcomingBtn.props.onClick);
    const mockchangeFilter = jest.fn(genresSelect.props.onChange);
    act(() => {
      mockFetchTopRatedMovies();
      mockFetchLatestMovies();
      mockFetchUpcomingMovies();
      mockchangeFilter({ target: { value: '123' } });
    });
    expect(mockFetchTopRatedMovies).toHaveBeenCalled();
    expect(mockFetchLatestMovies).toHaveBeenCalled();
    expect(mockFetchUpcomingMovies).toHaveBeenCalled();
    expect(mockchangeFilter).toHaveBeenCalled();
  });
  it('FilterPanel\'s defaultprops defined', () => {
    expect(FilterPanel.defaultProps.fetchTopRatedMovies).toBeDefined();
    expect(FilterPanel.defaultProps.fetchLatestMovies).toBeDefined();
    expect(FilterPanel.defaultProps.fetchUpcomingMovies).toBeDefined();
    expect(FilterPanel.defaultProps.changeFilter).toBeDefined();
  });
});
