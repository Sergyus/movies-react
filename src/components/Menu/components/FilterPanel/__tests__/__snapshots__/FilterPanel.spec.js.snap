// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterPanel component FilterPanel component renders correctly 1`] = `
<nav>
  <ul>
    <li
      className="navItem"
    >
      <button
        aria-label="Trending"
        className="btn btn__active"
        disabled={true}
        onClick={[Function]}
        type="button"
      >
        Trending
      </button>
    </li>
    <li
      className="navItem"
    >
      <button
        aria-label="Top Rated"
        className="btn"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        Top Rated
      </button>
    </li>
    <li
      className="navItem"
    >
      <button
        aria-label="Coming Soon"
        className="btn"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        Coming Soon
      </button>
    </li>
    <li
      className="navItem"
    >
      <select
        aria-label="Genres"
        className="selectGenre"
        name="selectGenre"
        onChange={[Function]}
        value=""
      >
        <option
          disabled={true}
          hidden={true}
          value=""
        >
          Genre
        </option>
      </select>
    </li>
  </ul>
</nav>
`;
